File: temperatura.py
import sys

input = sys.stdin
T = int(input.readline())
for t in range(T):
    N = int(input.readline())
    SEQ = input.readline()
    temp = 1
    result = "YES"
    for c in SEQ:
        if c == 'U':
            temp += 1
        else:
            temp -= 1
        if temp < 1 or temp > N:
            result = "NO"
            # break
    print(result)

File: temperatura.cpp
#include <iostream>
#include <string>
#include <limits>
#include <cstdio>

bool test(int n, int N, const std::string& seq) {
  bool ret = true;
  int temp = n;

  for (std::string::const_iterator it = seq.begin(); it != seq.end(); it++) {
    if (*it == 'U') {
      temp++;
    }
    else if (*it == 'D') {
      temp--;
    }
    if (temp < 1 || temp > N) {
      ret = false;
      break;
    }
  }
  return ret;
}

int main(int argc, char* argv[]) {
  std::string line;
  std::getline(std::cin, line);
  int T = std::stoi(line);

  while (T) {
    char c;
    int temp = 1;
    std::string result = "NO";
    std::getline(std::cin, line);
    const int N = std::stoi(line);
    int n = 1;
    std::getline(std::cin, line);
    while (n <= N) {
      if (test(n, N, line)) {
        result = "YES";
        break;
      }
      n++;
    }
    std::cout << result << std::endl;
    T--;
  }
  return 0;
}
